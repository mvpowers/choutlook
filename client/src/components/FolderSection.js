import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { FaChevronDown } from 'react-icons/lib/fa';

const Container = styled.div`
  display: flex;
  flex-direction: row;
  padding: 5px;
  padding-right: 100px;
  align-items: center;
  color: #333;
  font-weight: 400;
  font-size: 14px;

  &:hover {
    color: #000;
  }
`;

const Name = styled.div`
  display: flex;
  align-items: center;
  padding-right: 15px;
`;

const Icon = styled.div`
  padding: 15px;
`;

const FolderSection = ({ name }) => (
  <Container>
    <Icon>
      <FaChevronDown size={12} />
    </Icon>
    <Name>{name}</Name>
  </Container>
);

FolderSection.propTypes = {
  name: PropTypes.string.isRequired,
};

export default FolderSection;
