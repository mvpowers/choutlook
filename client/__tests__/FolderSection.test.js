import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FolderSection from '../src/components/FolderSection';

Enzyme.configure({ adapter: new Adapter() });

describe('FolderSection', () => {
  it('should be defined', () => {
    expect(FolderSection).toBeDefined();
  });
  it('should render correctly', () => {
    const tree = shallow(<FolderSection name="test" />);
    expect(tree).toMatchSnapshot();
  });
});
