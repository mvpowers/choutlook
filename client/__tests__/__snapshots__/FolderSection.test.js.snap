// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FolderSection should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FolderSection
    name="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <FolderSection__Icon>
          <FaChevronDown
            size={12}
          />
        </FolderSection__Icon>,
        <FolderSection__Name>
          test
        </FolderSection__Name>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <FaChevronDown
            size={12}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "size": 12,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <FolderSection__Icon>
            <FaChevronDown
              size={12}
            />
          </FolderSection__Icon>,
          <FolderSection__Name>
            test
          </FolderSection__Name>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <FaChevronDown
              size={12}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "size": 12,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
